### PR TITLE
Config file environment expansion

### LINKED ISSUE #280: Expand config file with environment variables
It would be a great feature if environment-variable placeholders like `${<ENVIRONMENT_VARIABLE>}` would be replaced by the environment variable set inside i.e. the grafana-agent container so that configuration values could be applied dynamically.

This would expand a config (assuming that the environment variable `LABEL_A` is set to the value `A_VALUE`) like:
```iml
prometheus:
  global:
    external_labels:
      labelA: ${LABEL_A}
```
to the finally parsed config:
```iml
prometheus:
  global:
    external_labels:
      labelA: A_VALUE
```

### PR DESCRIPTION
**Pull request by dcseifert**
#### PR Description 

It would be a great feature to expand the configuration file with environment variables like described in issue #280.

#### Which issue(s) this PR fixes 

Fixes #280 Expand config file with environment variables

#### PR Checklist

- [x] CHANGELOG updated 
- [ ] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by rfratto** [comment]
Thanks for working on this! 

Following up from our discussion earlier:

> Would disabling this feature by default (to avoid unexpected behavior) and enabling it via a switch be a valid option here?

I'd prefer it to work this way. The [Loki implementation](https://github.com/grafana/loki/pull/2837/files) of this (including the use of `github.com/drone/envsubst`) would be preferable. Can you make this behavior optional and turn it off by default? Using a flag to enable it should be more than enough. 

One other thing: please update the CHANGELOG and give yourself credit for working on this :)

**Comment by dcseifert** [comment]
Implementation is done and CHANGELOG is updated :)
I tried to stick to the Loki implementation as close as possible.

Looking forward to your review.

File: pkg/config/config.go [hunk 1]
```diff
@@ -65,19 +66,27 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
...
 // LoadBytes unmarshals a config from a buffer. Defaults are not
 // applied to the file and must be done manually if LoadBytes
 // is called directly.
-func LoadBytes(buf []byte, c *Config) error {
+func LoadBytes(buf []byte, expandEnvVars bool, c *Config) error {
+	// (Optionally) expand with environment variables
+	if expandEnvVars {
+		s, err := envsubst.EvalEnv(string(buf))
+		if err != nil {
+			return errors.Wrap(err, "unable to substitute config with environment variables")
```
Comment (rfratto) [inline pkg/config/config.go:85]:
```suggestion
			return fmt.Errorf("unable to substitute config with environment variables: %w", err)
```

Could this use `fmt.Errorf`? There's some legacy uses of `error.Wrap` here but I'm trying to just use Go's builtin wrapping moving forward.

**Review by rfratto — COMMENTED** [review]
One small nit but this looks great! Do you mind if I add a commit for documenting this in the configuration reference? You're also free to take a crack at it if you want.

**Comment by dcseifert** [comment]
No, feel free to add the docs!
I just skipped this step for now because I don't want to put the snippet from Loki into a wrong spot.

**Comment by rfratto** [comment]
Looks like the permission for me to add commits is disabled. I'll just merge this now and follow up with docs. Thanks again!

**Review by rfratto — APPROVED** [review]

**Pull request merged**